packages/beacon-node/src/chain/chain.ts

```diff
@@ -980,4 +981,11 @@ export class BeaconChain implements IBeaconChain {
       }
     }
   }
+
+  async getBlockRewards(block: allForks.FullOrBlindedBeaconBlock): Promise<BlockRewards> {
+    const preState = (await this.regen.getPreState(block, {dontTransferCache: true}, RegenCaller.restApi)).clone();
+    preState.slot = block.slot; // regen.getPreState guarantees pre_state of the same epoch but not the same slot
```

Comment:
why do you enforce `preState` to have the same slot of block?

Reply:
> why do you enforce `preState` to have the same slot of block?

Some old code. I just removed it